src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,113 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+    private StorageDaoImpl storageDao = new StorageDaoImpl();
+    private User user = new User();
+    private User userWithTheSameLogin1 = new User();
+    private User userWithTheSameLogin2 = new User();
+
+    @BeforeEach
+    void setUp() {
+        user.setLogin("login");
+        user.setPassword("password");
+        user.setAge(18);
+    }
+
+    @Test
+    void register_allDataCorrect_ok() {
+        User testUser = registrationService.register(user);
+        assertEquals(user, testUser);
+    }
+
+    @Test
+    void register_less6Characters_notOk() {
+        user.setPassword("qwerty");
+        assertThrows(RuntimeException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user.setPassword(null);
+        assertThrows(RuntimeException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_lessThan18_notOk() {
+        user.setAge(17);
+        assertThrows(RuntimeException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_more100_notOk() {
+        user.setAge(101);
+        assertThrows(RuntimeException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        user.setAge(-26);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
```

Comment:
You already have test that checks if age is undex 18

Reply:
Yeap...